### DISCUSSION TITLE
Conditional required properties in identity schemas

### DISCUSSION THREAD

**Discussion by SkypLabs** [ory/kratos#3019] [Q&A] [answered]
Hi,

I'd like to know if it's possible to use some keywords in the identity schemas to conditionally mark some attributes as required or not. For instance, with the `if-then-else` keywords as detailed [here](https://json-schema.org/understanding-json-schema/reference/conditionals.html#if-then-else).

All my attempts have failed so far despite my schemas being correctly validated by https://github.com/ory/jsonschema.

Thanks in advance!

**Comment by aeneasr** [comment, marked as answer]
We do support if-then-else in the schema, but the body parser and form generator do not really support these. It's better to stick to simple JSON schemas and build something using e.g. Ory Elements if you need more advanced logic!

**Reply by SkypLabs** [reply]
Thanks for your answer.

> It's better to stick to simple JSON schemas and build something using e.g. Ory Elements if you need more advanced logic!

I see JSON schemas in Kratos as a way to quickly design a registration logic rather than writing a separate custom service for it which would defeat the purpose of using JSON schemas in the first place. Ory Elements is mostly for front-end logic rather than enforcing it server-side, which is required for building reliable and secure software.

I think adding support for keywords like `enum` or `if-then-else` would be a very nice plus while keeping schemas relatively simple. It is just a matter of saying that if property `A` has as value `X`, property `B` is required, but if property `A` has as value `Y`, then `C` is required instead of `B`.

**Reply by SkypLabs** [reply]
After further testing, I can confirm that `enum` is correctly taken into account by Kratos:

```raw
I[#/traits/profileType] S[#/properties/traits/properties/profileType/enum] value must be one of "professional", "student"
```
